### ISSUE TITLE
[INDUSTRY_CORE] Update Semantic Models of IndustryCore 

### ISSUE THREAD

**Issue by jzbmw** [eclipse-tractusx/item-relationship-service#357]
**As** PO ,
**I want** that the IRS is able to handle the updated semantic models from the industry core,
**so that** we can provide a stable 24.05 Release .

## Blocked 
This is blocked until new test data is available

## Link 
- child of eclipse-tractusx/sig-release#219
- blocked by #367 
- relates to https://github.com/eclipse-tractusx/item-relationship-service/issues/400

## Hints / Details

- Industry Core Confluence: https://confluence.catena-x.net/display/BDPQ/%5BIC%5D+Industry+Core
- https://github.com/catenax-eV/product-standardization-prod/blob/CX-0126-IndustryCorePartType-v1.0.0/standards/CX-0126-IndustryCorePartType/1.0.0/CX-0126-Industry-Core-Part-Type.pdf
- https://github.com/catenax-eV/product-standardization-prod/blob/CX-0127-IndustryCorePartInstance-v1.0.0/standards/CX-0127-IndustryCorePartInstance/1.0.0/CX%20-%200127%20Industry%20Core%20Part%20Instance%201.0.0.pdf

### Acceptance Criteria
- [ ] Aspect Models have been updated
- [ ] JIS, Batch, Software, PartType, PartInstance
- [ ] IRS supports the following mandatory aspect models 
- [ ] IRS supports the following optional aspect models 
- [ ]  Add tests for new aspect models, including shared PartSiteInformationAsBuilt (covered with Mock infrastructure)
- [ ] Update Compatibility Matrix including new supported semantic models  
- [ ] Report evidence that industry core semantic models are supported by IRS version 

| Checked |Artefact| Version | Availability                                                 | link                                                                                                                                              |
|----|---|---------|--------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
| | AAS |         |                                                              |                                                                                                                                                   |
| [x] https://jira.catena-x.net/browse/TRI-893 |PartAsPlanned | 2.0.0   | Mandatory             | [PartAsPlanned 2.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.part_as_planned/2.0.0)                        |
| [x] | SingleLevelBomAsPlanned| 3.0.0   | Mandatory              | [SingleLevelBomAsPlanned 3.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.single_level_bom_as_planned/3.0.0)  |
| [ ]  INT Test in ERROR https://jira.catena-x.net/browse/TRI-767| SerialPart| 3.0.0   | Mandatory                                                    | [SerialPart 3.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.serial_part/3.0.0)                           |
|  [ ]  INT Test in ERROR https://jira.catena-x.net/browse/TRI-767 | Batch| 3.0.0   | Mandatory                                                    | [Batch 3.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.batch/3.0.0)                                          |
|  [ ]  INT Test in ERROR https://jira.catena-x.net/browse/TRI-767 JustInSequencePart| 3.0.0   | Mandatory                                                     | [JustInSequencePart 3.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.just_in_sequence_part/3.0.0)             |
|  [ ]  INT Test in ERROR https://jira.catena-x.net/browse/TRI-767 SingleLevelBomAsBuilt| 3.0.0   | Mandatory                                                    | [SingleLevelBomAsBuilt 3.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.single_level_bom_as_built/3.0.0)      |
|  [ ]  INT Test in ERROR https://jira.catena-x.net/browse/TRI-767 PartSiteInformationAsBuilt| 1.0.0   | Integrated in Aspects JustInSequencePart / Batch / SerialPart | [PartSiteInformationAsBuilt 1.0.0](https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.shared.part_site_information_as_built/1.0.0)                         |

## Out of Scope

- Preparing Testdata.

**Comment by mkanal** [comment]
@jzbmw currently test data set covers all expected aspect models - this story is ready to implement.

**Comment by ds-jhartmann** [comment]
@jzbmw @mkanal this is marked as DISCUSSION_NEEDED. Are there still open topics?

**Comment by ds-jhartmann** [comment]
This is depending on the latest testdata set which will be updated in https://github.com/eclipse-tractusx/item-relationship-service/issues/488

**Comment by dsmf** [comment]
blocked by #488

**Comment by mkanal** [comment]
> blocked by #488

Please add some more context. Please add BLOCKED label only after an alignment in daily. Thank you very much,

**Comment by mkanal** [comment]
Removed "BLOCKED" label as there has to be alignment first to get a common understanding about the underlying impediment.

**Comment by mkanal** [comment]
Hi @dsmf 
thank you for addressing the issue. The dependency was regarding Industry Core standard 1.0.0. Meanwhile, there is Industry Core with the new aspect versions 3.0.0. So we should directly update to this version. The semantic team has although updated this versions directly in the testdata set. If there is still an impediment, please align on this directly p2p.

**Comment by mkanal** [comment]
BLOCKED is removed as soon testdata will be updated.

**Comment by ds-jhartmann** [comment]
Semantic Hub does not contain `urn:samm:io.catenax.part_as_planned:2.0.0#PartAsPlanned`
Testdata set does not contain `urn:samm:io.catenax.shared.part_site_information_as_built:1.0.0#PartSiteInformationAsBuilt`
Do you have further Information on this topic? @mkanal @jzbmw

**Comment by ds-jhartmann** [comment]
## Outcome
- Created new Cucumber Tests to cover the new Semantic Models 
  - TRI-1913 "3.0.0#JustInSequencePart", "3.0.0#SerialPart" with "1.0.0#PartSiteInformationAsBuilt"
  - TRI-1914 "3.0.0#SingleLevelBomAsBuilt", "3.0.0#Batch"
  - TRI-1915 "3.0.0#SingleLevelBomAsPlanned", "2.0.0#PartAsPlanned"
- Fixed provisioning of local models 
- Switched to provided Semantic Models for the DEV environment. 
- Triggered SemHub Team to add urn:samm:io.catenax.part_as_planned:2.0.0#PartAsPlanned to INT Semantic Hub https://github.com/eclipse-tractusx/sldt-semantic-hub/issues/270
- created PBI to combine locally provided models and Semantic Hub models https://github.com/eclipse-tractusx/item-relationship-service/issues/526

**Comment by dsmf** [comment]
- https://github.com/catenax-ng/tx-item-relationship-service/pull/857
- https://github.com/catenax-ng/tx-item-relationship-service/pull/858

**Comment by mkanal** [comment]
@dsmf  @ds-jhartmann @ds-kgassner 
INT Tests https://jira.catena-x.net/browse/TRI-767 https://jira.catena-x.net/browse/TRI-1913  https://jira.catena-x.net/browse/TRI-1914 https://jira.catena-x.net/browse/TRI-704 must be executed successful then this is done.

**Comment by ds-jhartmann** [comment]
@mkanal TRI-767 and TRI-704 are not relevant for this issue since they request a entire vehicle. As seen in this test execution [TRI-1927](https://jira.catena-x.net/browse/TRI-1927), the cucumber test executions were successful at the time.

**Comment by mkanal** [comment]
@ds-jhartmann  I could not see test evidence. I will set this to test again. 
![image](https://github.com/eclipse-tractusx/item-relationship-service/assets/100209308/be099be2-c95c-4117-82cc-c3ffe9754f46)

**Comment by ds-jhartmann** [comment]
As seen here, all relevant Cucumber Tests are successful: https://jira.catena-x.net/browse/TRI-2013
![image](https://github.com/eclipse-tractusx/item-relationship-service/assets/57985712/033e607b-5d4d-46d3-8ab4-19ad487b6090)

**Comment by mkanal** [comment]
All tests run successfully. PO acceptance in behalf of @jzbmw
